src/tools/clippy/clippy_utils/src/ty.rs

```diff
@@ -223,9 +224,10 @@ pub fn implements_trait_with_env<'tcx>(
     param_env: ParamEnv<'tcx>,
     ty: Ty<'tcx>,
     trait_id: DefId,
+    callee_id: DefId,
```

Comment:
@fee1-dead (cc: @compiler-errors) I am having difficulty understanding what this additional required `callee_id` parameter is/does.

A comment elsewhere in this PR suggests it is an "effect arg". Moreover, I found this doc comment for `with_opt_host_effect_param`: https://github.com/rust-lang/rust/blob/d73bd3fb3ba312f3e6b5af4d56d1161d37b71620/compiler/rustc_middle/src/ty/util.rs#L860

But I can't find any documentation (in the Rustc Dev Guide) on what an "effect arg/param" is, or why/when it is needed.

Could you please point me to something that could help me to get a better understanding? Thank you.

Reply:
I've hidden the last comment since it could be wrong. I will take a look maybe tomorrow.